README.md

```diff
@@ -54,6 +58,90 @@ jobs:
           push: true
 ```
 
+### Git context
```

Comment:
In https://github.com/docker/buildkit-syft-scanner/pull/90/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR71 it looks like "files" need to be set differently in this case. Could this be avoided?

`{{defaultContext}}` is something Github provides?

Any path to make this default and `source: "."` the alternative? Or is it too incompatible?

Reply:
> In https://github.com/docker/buildkit-syft-scanner/pull/90/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR71 it looks like "files" need to be set differently in this case. Could this be avoided?

Yes I think we could detect if source is remote and automatically prepend any files with `cwd://`.

> `{{defaultContext}}` is something Github provides?

No this is provided by us through the actions toolkit: https://github.com/docker/actions-toolkit/blob/5430d017a5f0d7c989b91856df03f754135f2627/src/context.ts#L52-L54 so we can return a proper Git URI.

> Any path to make this default and `source: "."` the alternative? Or is it too incompatible?

That would be a breaking change if we switch to git context by default unfortunately but I don't mind cutting a major release.

Reply:
> Yes I think we could detect if source is remote and automatically prepend any files with `cwd://`.

Actually no it should be done directly in the metadata-action where we always set `cwd://` if buildx version >= 0.12.

Reply:
Opened https://github.com/docker/metadata-action/pull/370

Reply:
> That would be a breaking change if we switch to git context by default unfortunately but I don't mind cutting a major release.

Made changes to default to git context. `source: .` means we want to build from local. Updated docs in README as well.

Reply:
> t looks like "files" need to be set differently in this case

Does this new solution work with both metadata-action and files selected from the remote repository?

What are the cases that would break when people upgrade to new version?

Reply:
> What are the cases that would break when people upgrade to new version?

Any file mutation in the steps that precede the build step will be ignored with Git context

> Does this new solution work with both metadata-action and files selected from the remote repository?

Yes since Buildx 0.12 (https://github.com/docker/buildx/pull/1838)